From 3c3bab59389ff88336851001ffbedb2d8c6bc572 Mon Sep 17 00:00:00 2001
From: Haohui Mai <wheat9@apache.org>
Date: Mon, 27 Oct 2014 16:22:19 -0700
Subject: [PATCH 1059/2848] HDFS-5928. Show namespace and namenode ID on NN
 dfshealth page. Contributed by Siqi Li.

(cherry picked from commit 0e056246180f87ebcf3970260262baee34d75a92)

Change-Id: I852842557f29af103a36636b9e2259ede6b9c545
---
 .../src/main/webapps/hdfs/dfshealth.html           |    9 +++++++--
 .../hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js |   17 +++++++++++++++++
 2 files changed, 24 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
index d12f991..73058e8 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.html
@@ -123,15 +123,20 @@
 {/nn}
 
 <div class="page-header"><h1>Overview {#nnstat}<small>'{HostAndPort}' ({State})</small>{/nnstat}</h1></div>
-{#nn}
+
 <table class="table table-bordered table-striped">
+{#HAInfo}
+  <tr><th>Namespace:</th><td>{Namespace}</td></tr>
+  <tr><th>Namenode ID:</th><td>{NamenodeID}</td></tr>
+{/HAInfo}
+{#nn}
   <tr><th>Started:</th><td>{NNStarted}</td></tr>
   <tr><th>Version:</th><td>{Version}</td></tr>
   <tr><th>Compiled:</th><td>{CompileInfo}</td></tr>
   <tr><th>Cluster ID:</th><td>{ClusterId}</td></tr>
   <tr><th>Block Pool ID:</th><td>{BlockPoolId}</td></tr>
-</table>
 {/nn}
+</table>
 
 <div class="page-header"><h1>Summary</h1></div>
 <p>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js
index 04e43d2..ea1ee02 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js
@@ -55,6 +55,23 @@
 
     var data = {};
 
+    $.ajax({'url': '/conf', 'dataType': 'xml', 'async': false}).done(
+      function(d) {
+        var $xml = $(d);
+        var namespace, nnId;
+        $xml.find('property').each(function(idx,v) {
+          if ($(v).find('name').text() === 'dfs.nameservice.id') {
+            namespace = $(v).find('value').text();
+          }
+          if ($(v).find('name').text() === 'dfs.ha.namenode.id') {
+            nnId = $(v).find('value').text();
+          }
+        });
+        if (namespace && nnId) {
+          data['HAInfo'] = {"Namespace": namespace, "NamenodeID": nnId};
+        }
+    });
+
     // Workarounds for the fact that JMXJsonServlet returns non-standard JSON strings
     function workaround(nn) {
       nn.JournalTransactionInfo = JSON.parse(nn.JournalTransactionInfo);
-- 
1.7.9.5

